extensions/default/src/init.ts

```diff
@@ -38,14 +47,53 @@ export default function init({ servicesManager, configuration = {} }): void {
   });
 
   // Stores a map from the to be applied hanging protocols `<activeStudyUID>:<protocolId>`
-  // to the previously applied hanging protolStageIndexMap key, in order to toggle
+  // to the previously applied hanging protocolStageIndexMap key, in order to toggle
   // off the applied protocol and remember the old state.
   stateSyncService.register('toggleHangingProtocol', { clearOnModeExit: true });
 
   // Stores the viewports by `rows-cols` position so that when the layout
   // changes numRows and numCols, the viewports can be remembers and then replaced
   // afterwards.
   stateSyncService.register('viewportsByPosition', { clearOnModeExit: true });
+
+  // Function to process and subscribe to events for a given set of commands and listeners
+  const subscribeToEvents = listeners => {
+    Object.entries(listeners).forEach(([event, commands]) => {
+      const supportedEvents = [
+        viewportGridService.EVENTS.ACTIVE_VIEWPORT_ID_CHANGED,
+        viewportGridService.EVENTS.VIEWPORTS_READY,
+      ];
+
+      if (supportedEvents.includes(event)) {
+        viewportGridService.subscribe(event, eventData => {
+          const viewportId = eventData?.viewportId ?? viewportGridService.getActiveViewportId();
+
+          commandsManager.run(commands, { viewportId });
```

Comment:
You should be aware that the viewport grid service fires events BEFORE the viewport actually gets updated, so if your new events depend on the viewport having already changed, it won't respond with what you expect.